omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/AdministrationLinksResource1_8.java

```diff
@@ -0,0 +1,174 @@
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8;
+
+import io.swagger.models.Model;
+import io.swagger.models.ModelImpl;
+import io.swagger.models.properties.ArrayProperty;
+import io.swagger.models.properties.ObjectProperty;
+import io.swagger.models.properties.StringProperty;
+import org.openmrs.module.Extension;
+import org.openmrs.module.ModuleFactory;
+import org.openmrs.module.web.extension.AdministrationSectionExt;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.PropertyGetter;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.RefRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingReadableResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ObjectNotFoundException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.module.webservices.rest.web.v1_0.wrapper.openmrs1_8.ModuleAdministrationLinks1_8;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Resource(name = RestConstants.VERSION_1 + "/administration-links", supportedClass = ModuleAdministrationLinks1_8.class,
+		supportedOpenmrsVersions = { "1.8.*", "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*", "2.2.*", "2.3.*",
+				"2.4.*", "2.5.*" })
+public class AdministrationLinksResource1_8 extends BaseDelegatingReadableResource<ModuleAdministrationLinks1_8> {
+
+	public static final String ADMIN_LIST_POINT_ID = "org.openmrs.admin.list";
+
+	public static final String MODULE_TITLE = "moduleTitle";
+
+	public static final String LINKS = "administrationLinks";
+
+	@Override
+	public ModuleAdministrationLinks1_8 newDelegate() {
+		return null;
+	}
+
+	@Override
+	public ModuleAdministrationLinks1_8 getByUniqueId(String moduleId) {
+		// Assumes that moduleId is an id of a module that user wants to get admin links for.
+		// AdministrationLinksResource has no id by itself as it's not an OpenMRS data object.
+
+		ModuleAdministrationLinks1_8 administrationLinks = getAdministrationLinksForModule(moduleId);
+		if (administrationLinks == null) {
+			throw new ObjectNotFoundException();
+		}
+
+		return getAdministrationLinksForModule(moduleId);
```

Comment:
Why are you calling this twice?

Reply:
Good catch, will change in the next commit :)